feathr_project/feathr/utils/_envvariableutil.py

```diff
@@ -5,87 +5,116 @@
 from azure.core.exceptions import ResourceNotFoundError
 
 class _EnvVaraibleUtil(object):
-    def __init__(self, config_path):
+    """A utility class to read config variables from environment variables.
+    If use_env_vars set to False, it will return the default value from the config file.
+    """
+    # Azure Key Vault name to retrieve environment variables
+    akv_name: str = None
+
+    def __init__(self, config_path: str, use_env_vars: bool = True):
+        """Initialize the utility class.
+
+        Args:
+            config_path: Config file path.
+            use_env_vars (optional): Whether to use environment variables. Defaults to True.
+        """
         self.config_path = config_path
-        # Set to none first to avoid invalid reference
-        self.akv_name = None
-        self.akv_name = self.get_environment_variable_with_default( 'secrets', 'azure_key_vault', 'name')
+        self.use_env_vars = use_env_vars
+        self.akv_name = self.get_environment_variable_with_default("secrets", "azure_key_vault", "name")
         self.akv_client = AzureKeyVaultClient(self.akv_name) if self.akv_name else None
 
-    def get_environment_variable_with_default(self, *args):
+    def get_environment_variable_with_default(self, *args) -> str:
         """Gets the environment variable for the variable key.
+        If use_env_vars set to False, it will return the default value from the config file.
+
         Args:
             *args: list of keys in feathr_config.yaml file
-        Return:
-            A environment variable for the variable key. It will retrieve the value of the environment variables in the following order:
+
+        Returns:
+            An environment variable for the variable key. It will retrieve the value of the environment variables in the following order:
```

Comment:
Should we change the 'Returns' comments here? I guess if 'use_env_vars' is False, we will not return the environment variable when the key is set.

Reply:
yes. I'll update the docstring accordingly. Thank you!